### PR TITLE
fix(deps): Upgrade dependencies to address CVE-2023-5077 and upgrade to Go 1.20

### PR DESCRIPTION
**Pull request by dbellinghoven**
Upgrades the github.com/hashicorp/vault to the latest version (v1.15.0) to address a vulnerability (CVE-2023-5077).

Upgrades Go to v1.20.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,214 +1,292 @@
...
+	github.com/google/go-cmp v0.5.9
+	github.com/hashicorp/go-hclog v1.5.0
 	github.com/hashicorp/go-uuid v1.0.3
-	github.com/hashicorp/vault v1.11.4
-	github.com/hashicorp/vault/api v1.7.2
-	github.com/hashicorp/vault/sdk v0.5.3-0.20220826205147-7caf353e8124
+	github.com/hashicorp/hcl v1.0.1-vault-5
+	github.com/hashicorp/vault v1.15.0
+	github.com/hashicorp/vault/api v1.10.0
+	github.com/hashicorp/vault/sdk v0.10.1-0.20230922151601-0fa36a36ae1b
```
Comment (rliebz) [inline go.mod:13]: Just curious, how'd you find the relevant commit?

File: Dockerfile [hunk 2]
```diff
@@ -11,7 +11,7 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
-FROM golang:1.17.5-alpine3.14
+FROM golang:1.20.8-alpine3.18
```
Comment (rliebz) [inline Dockerfile:14]: Any reason to choose 1.20 over 1.21?

(1.20 is still supported so it's definitely not an issue)

**Review by rliebz — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (dbellinghoven) [reply go.mod:13]: I actually just picked the latest commit that changed files in the `github.com/hashicorp/vault/sdk` package 
https://github.com/hashicorp/vault/tree/main/sdk

File: Dockerfile (hunk 2 above)
Comment (dbellinghoven) [reply Dockerfile:14]: Good question - I chose 1.20 just because I am still on 1.20 locally 😂

**Review by rliebz — DISMISSED** [review]

**Review by rliebz — APPROVED** [review]

**Review by alexdulin — APPROVED** [review]

**Pull request merged**